src/litdata/streaming/dataloader.py

```diff
@@ -700,13 +701,23 @@ def load_state_dict(self, obj: Dict[str, Any]) -> None:
 
         # Inform we are resuming and disable resetting the StreamingDataLoader state.
         # This is toggle back to False when the `__iter__` method of the StreamingDataLoader completes.
-        self.restore = True
+        # self.restore = True
 
         if isinstance(self.dataset, CombinedStreamingDataset):
             self.dataset._set_use_streaming_dataloader(True)
             self.dataset.load_state_dict(obj)
+
+            # Inform that the dataloader is resuming.
+            # TODO: Check if the number of samples yielded is less than the length of the dataset.
+            # Also, len is not available for CombinedStreamingDataset incase of provided weights.
+            self.restore = True
```

Comment:
Hi @tchaton,

Here, I was trying to separate the logic to enable resuming for each type of dataset but got a bit stuck on how to handle the CombinedStreamingDataset. Since I won't have the length of the dataset to compare with the provided weights, I'm wondering if you have any suggestions on how to approach this issue.

Thank you for your help!

Reply:
**Fixes related to `CombinedStreamingDataset` will be included in a separate PR referencing a new issue #331.**

cc: @tchaton